docs/rules/block-attributes-order.md

```diff
@@ -0,0 +1,92 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/block-attributes-order
+description: enforce order of block attributes
+since: v4.3.0
+---
+# vue/block-attributes-order
+
+> enforce order of block attributes
+
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+This rule aims to enforce ordering of block attributes.
+
+### The default order
```

Comment:
I think we need to discuss option formats.

If we want to be more granular and configurable by the user, I suggest using an array of objects with elements and ordering, like this:

```js
{'vue/block-attributes-order': ['error',
  [
    { element: 'script', order: ['setup, 'lang', 'src'] },
    { element: 'template', order: ['functional', 'lang', 'src'] },
    { element: 'style', order: ['scoped', 'module', 'lang', 'src'] },
    { element: 'i18n', order: ['global', 'locale', 'lang', 'src'] },
  ]
]}
```

You can use CSS selectors for the `element` property. This is similar to the [vue/component-tags-order option](https://eslint.vuejs.org/rules/component-tags-order.html#options).
The order option specifies the order of the attributes. It can use attribute names, regular expressions and their arrays, and negation. This is similar to the [svelte/sort-attributes option](https://ota-meshi.github.io/eslint-plugin-svelte/rules/sort-attributes/#options).

If you want lang and src to always be at the back:

```js
{'vue/block-attributes-order': ['error',
  [
    { element: '*', order: [['/.*/', '!/^(?:src|lang)$/'], 'lang', 'src'] }
  ]
]}
```

Patterns can be used to correct some order even for unknown blocks and unknown attributes.

What do you think?